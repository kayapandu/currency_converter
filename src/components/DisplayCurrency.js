import React from 'react';
import { Col, Button } from 'reactstrap';

class DisplayCurrency extends React.Component {
    constructor(props){
        super(props);
    }

    render(){
    const { name, ratesValue, nowRates } = this.props;
        return(
            <Col style={styles.boxCurrency}>
                <Col xs="10">
                    <Col style={{
                        display: "flex",
                        flexDirection: "row",
                        flexWrap: "wrap",
                        justifyContent: "space-between",

                        }}
                    >
                        <Col xs="6" style={{fontWeight: "bold", fontSize: "24px"}}>
                            {name}
                        </Col>
                        <Col xs="6"  style={{fontWeight: "bold", fontSize: "24px", textAlign: "right"}}>
                            {ratesValue}
                        </Col>
                    </Col>
                    <Col style={{padding: "10px", marginLeft: "20px"}}>
                    {`1 USD = ${name} ${nowRates}`}
                    </Col>
                </Col>
                <Col style={{ alignItems: "stretch", alignSelf: "center", textAlign: "center"}}>
                    <Button style={{width: "100%"}} onClick={() => this.props.removeRates(name)}>Remove</Button>
                </Col>
            </Col>
        );
    }

}

export default DisplayCurrency;

const styles = {
    boxCurrency : {
        display: "flex",
        flexDirection: "row",
        flexWrap: "wrap",
        border: "1px black solid",
        borderRadius: "0.5em",
        margin: "5px",
        padding: "10px"
    }
}