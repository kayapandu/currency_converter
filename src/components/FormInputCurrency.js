import React from 'react';
import { InputGroup, InputGroupAddon, Input } from 'reactstrap';

class FormInputCurrency extends React.Component {

    render(){
        const { value, updateValue } = this.props;
        return(
            <InputGroup>
                <InputGroupAddon size="lg" addonType="prepend">USD</InputGroupAddon>
                <Input type="number" onChange={updateValue} placeholder="Input Amount" value={value} />
            </InputGroup>
        )
    }
}

export default FormInputCurrency;