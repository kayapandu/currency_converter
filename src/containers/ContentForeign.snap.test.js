import React from 'react';
import renderer from 'react-test-renderer';

import FormInputCurrency from '../components/FormInputCurrency';
import DisplayCurrency from '../components/DisplayCurrency';

describe('render correctly', () => {
    test('should render FormInputCurrency', () => {
        const component = renderer.create(
            <FormInputCurrency/>
        )
        
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should render DisplayCurrency', () => {
        const component = renderer.create(
            <DisplayCurrency/>
        )

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})