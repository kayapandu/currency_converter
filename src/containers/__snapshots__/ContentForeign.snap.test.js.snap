// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly should render DisplayCurrency 1`] = `
<div
  className="col"
  style={
    Object {
      "border": "1px black solid",
      "borderRadius": "0.5em",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "margin": "5px",
      "padding": "10px",
    }
  }
>
  <div
    className="col-10"
  >
    <div
      className="col"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        className="col-6"
        style={
          Object {
            "fontSize": "24px",
            "fontWeight": "bold",
          }
        }
      />
      <div
        className="col-6"
        style={
          Object {
            "fontSize": "24px",
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
      />
    </div>
    <div
      className="col"
      style={
        Object {
          "marginLeft": "20px",
          "padding": "10px",
        }
      }
    >
      1 USD = undefined undefined
    </div>
  </div>
  <div
    className="col"
    style={
      Object {
        "alignItems": "stretch",
        "alignSelf": "center",
        "textAlign": "center",
      }
    }
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      type="button"
    >
      Remove
    </button>
  </div>
</div>
`;

exports[`render correctly should render FormInputCurrency 1`] = `
<div
  className="input-group"
>
  <div
    className="input-group-prepend"
    size="lg"
  >
    <span
      className="input-group-text"
    >
      USD
    </span>
  </div>
  <input
    className="form-control"
    placeholder="Input Amount"
    type="number"
  />
</div>
`;
